### PR TITLE
fix bug html not working

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
solve issue #59 

Turns out there isn't any problem with html rendering. The problem that is preventing html from rendering is the wrong format of head content (it should be a list but it was a string).

The rest of the pr are minor improvements here and there and a few tests for both markdown and html rendering with a bit of styling

### CODE REVIEW

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -839,7 +839,11 @@ def run(self, run_server: bool = True, run_in_thread: bool = False, **kwargs) ->
                 print(f" * NGROK Public Url: {http_tunnel.public_url}")
 
         # Save all local variables of the parent frame (usually __main__)
-        self.__locals_bind = t.cast(FrameType, t.cast(FrameType, inspect.currentframe()).f_back).f_locals
+        if "locals_bind" in kwargs and isinstance(kwargs["locals_bind"], dict):
+            self.__locals_bind = kwargs["locals_bind"]
+            warnings.warn("Gui is using custom locals_bind. Make sure that you know what you are doing")
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py]: I would prefer something like "Doing this or that may fail."

File: tests/taipy/gui/helpers.py [hunk 2]
```diff
@@ -80,3 +84,13 @@ def port_check():
             return True
         except:
             return False
+
+    @staticmethod
+    def run_e2e(gui, **kwargs):
```
Comment (FabienLelaquais) [inline tests/taipy/gui/helpers.py:89]: I like that!

**Review by FabienLelaquais — APPROVED** [review]
Looks great!

File: tests/taipy/gui/helpers.py (hunk 2 above)
Comment (dinhlongviolin1) [reply tests/taipy/gui/helpers.py:89]: I know cool trick :D

File: taipy/gui/gui.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: Can you type out the whole sentence like usual :D

File: taipy/gui/gui.py [hunk 3]
```diff
@@ -839,7 +842,11 @@ def run(self, run_server: bool = True, run_in_thread: bool = False, **kwargs) ->
                 print(f" * NGROK Public Url: {http_tunnel.public_url}")
 
         # Save all local variables of the parent frame (usually __main__)
-        self.__locals_bind = t.cast(FrameType, t.cast(FrameType, inspect.currentframe()).f_back).f_locals
+        if "locals_bind" in kwargs and isinstance(kwargs["locals_bind"], dict):
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py]: This is quite bold ;-)
but it has **huge** potential.
side note: `if kwargs.get("locals_bind"), dict):` would achieve the same test

File: taipy/gui/renderers/_html/parser.py [hunk 4]
```diff
@@ -118,4 +118,7 @@ def set_value(self, value: str) -> bool:
     def parse(self, gui) -> t.Tuple[str, str]:
         for k, v in self.properties.items():
             self.properties[k] = v if v is not None else "true"
+        # allow usage of 'class' property in html taipy tag
+        if "class" in self.properties and "classname" not in self.properties:
```
Comment (FredLL-Avaiga) [inline taipy/gui/renderers/_html/parser.py:122]: nice touch

File: taipy/gui/state.py [hunk 5]
```diff
@@ -33,8 +33,7 @@ def __getattribute__(self, name: str) -> t.Any:
             raise AttributeError(f"Variable '{name}' is not defined.")
         if not hasattr(gui._bindings(), name):
             gui._bind_var(name)
-        val = getattr(gui._bindings(), name)
-        return val
+        return getattr(gui._bindings(), name)
```
Comment (FredLL-Avaiga) [inline taipy/gui/state.py:36]: 👍

File: tests/taipy/gui/e2e/renderers/test_html_rendering.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+def test_html_render_bind_assets(page: "Page", gui: Gui, helpers):
+    gui.add_pages(pages=f"{Path(Path(__file__).parent.resolve())}{os.path.sep}test-assets")
+    helpers.run_e2e(gui)
+    assert ".taipy-text" in urlopen("http://127.0.0.1:5000/test-assets/style/style.css/").read().decode("utf-8")
+    page.goto("/page1")
+    page.expect_websocket()
+    page.wait_for_selector("#text1")
+    retry = 0
+    while (
```
Comment (FredLL-Avaiga) [inline tests/taipy/gui/e2e/renderers/test_html_rendering.py:55]: Would be nice to find a way to ask playwright to wait for style to "appear"

**Review by FredLL-Avaiga — APPROVED** [review]
🥇

File: tests/taipy/gui/e2e/renderers/test_html_rendering.py (hunk 6 above)
Comment (dinhlongviolin1) [reply tests/taipy/gui/e2e/renderers/test_html_rendering.py:55]: it is a good idea. Let me have a look at it

File: taipy/gui/gui.py (hunk 1 above)
Comment (FabienLelaquais) [reply taipy/gui/gui.py]: warnings.warn("Caution: the Gui instance is using a custom 'locals_bind' setting.")

File: taipy/gui/renderers/_html/parser.py (hunk 4 above)
Comment (dinhlongviolin1) [reply taipy/gui/renderers/_html/parser.py:122]: thanks, writing tests gave me the idea :D

File: taipy/gui/gui.py (hunk 3 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: `if kwargs.get("locals_bind"), dict):` is cool but it is missing an opening bracket right

File: taipy/gui/gui.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: thanks!

File: taipy/gui/gui.py (hunk 3 above)
Comment (FredLL-Avaiga) [reply taipy/gui/gui.py]: yes but you get the idea, the full text would be:
`if isinstance(kwargs.get("locals_bind"), dict):`

Comment (dinhlongviolin1) [reply taipy/gui/gui.py]: oh oh ok make sense now. I thought it was some magic that I didn't know about. Thanks Fred

**Pull request merged**